NumLib/TimeStepping/Algorithms/EvolutionaryPIDcontroller.cpp

```diff
@@ -104,6 +109,28 @@ bool EvolutionaryPIDcontroller::next(const double solution_error)
     return true;
 }
 
+double EvolutionaryPIDcontroller::limitStepSize(
+    const double h_new, const double h_n,
+    const bool previous_step_accepted) const
+{
+    const double h_in_range = std::max(_h_min, std::min(h_new, _h_max));
+    double limited_h =
+        std::max(_rel_h_min * h_n, std::min(h_in_range, _rel_h_max * h_n));
```

Comment:
It would be nice to have some docu about the limits either here or in a function's documentation.
Something along the following lines:
"At first the absolute limits are enforced h_min,..h_max, then the maximum/minimum timestep change is within limits..."